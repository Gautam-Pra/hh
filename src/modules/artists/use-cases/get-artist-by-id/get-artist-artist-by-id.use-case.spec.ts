import { beforeAll, describe, expect, test } from 'vitest'
import { GetArtistByIdUseCase } from '#modules/artists/use-cases'

describe('GetArtistById', () => {
  let getArtistByIdUseCase: GetArtistByIdUseCase

  beforeAll(() => {
    getArtistByIdUseCase = new GetArtistByIdUseCase()
  })

  test('should get artist by id and return an artist', async () => {
    const artist = await getArtistByIdUseCase.execute({
      artistId: '1274170',
      page: 1,
      songCount: 5,
      albumCount: 5,
      sortBy: 'popularity',
      sortOrder: 'desc'
    })

    expect(artist).toMatchSnapshot({
      ...artist,
      followerCount: expect.any(Number),
      topSongs: expect.any(Array),
      singles: expect.any(Array),
      topAlbums: expect.any(Array)
    })
  })
})
