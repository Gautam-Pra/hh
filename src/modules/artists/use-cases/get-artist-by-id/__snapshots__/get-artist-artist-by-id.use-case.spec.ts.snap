// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GetArtistById > should get artist by id and return an artist 1`] = `
{
  "availableLanguages": [
    "english",
    "unknown",
    "spanish",
    "instrumental",
    "italian",
    "french",
    "portuguese",
    "hebrew",
    "dutch",
    "korean",
  ],
  "bio": [],
  "dob": "",
  "dominantLanguage": "english",
  "dominantType": "music director",
  "fanCount": "706746",
  "fb": "",
  "followerCount": Any<Number>,
  "id": "1274170",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
    },
  ],
  "isRadioPresent": true,
  "isVerified": true,
  "name": "Dua Lipa",
  "similarArtists": [],
  "singles": Any<Array>,
  "topAlbums": Any<Array>,
  "topSongs": Any<Array>,
  "twitter": "",
  "type": "artist",
  "url": "https://www.jiosaavn.com/artist/dua-lipa/r-OWIKgpX2I_",
  "wiki": "",
}
`;
